src/SharpCompress/Compressors/Deflate/GZipStream.cs

```diff
@@ -384,16 +384,7 @@ public override void Write(byte[] buffer, int offset, int count)
                 if (_fileName.Contains("\\"))
                 {
                     // trim any leading path
-                    int length = _fileName.Length;
-                    int num = length;
-                    while (--num >= 0)
-                    {
-                        char c = _fileName[num];
-                        if (c == '\\')
-                        {
-                            _fileName = _fileName.Substring(num + 1, length - num - 1);
```

Comment:
the main issue was here - missing break after this line. I decided to simplify code though.